### PR TITLE
173062409 page layouts

### PR DESCRIPTION
**Pull request by mklewandowski**
This PR adds the basic activity page layout types.  It includes:
- new mock-data for testing
- full width, responsive, 60-40, 40-60 layouts
- navigation buttons at the bottom of the activity page
- pinned interactives when scrolling

### CODE REVIEW

File: src/components/activity-header/activity-nav-header.scss [hunk 1]
```diff
@@ -10,6 +10,11 @@
   background-color: #414546;
   color: white;
   font-size: 18px;
+  margin: 0 20px 0 20px;
```
Comment (kswenson) [inline src/components/activity-header/activity-nav-header.scss:13]:
```suggestion
  margin: 0 20px;
```
DRY

File: src/components/activity-page/activity-page-content.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    this.state = {
+      scrollOffset: 0
+    };
+  }
+
+  render() {
+    const { isFirstActivityPage, isLastActivityPage, page, totalPreviousQuestions } = this.props;
+    const { scrollOffset } = this.state;
+    const fullWidth = page.layout === PageLayouts.Responsive;
+    const vertical = page.layout === PageLayouts.FullWidth;
```
Comment (kswenson) [inline src/components/activity-page/activity-page-content.tsx:35]: A priori, it seems a bit odd that `PageLayouts.FullWidth` does not determine the `fullWidth` value. Perhaps a comment to clarify the contextual meanings of the terms here?

File: src/components/activity-page/activity-page-content.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    const renderPrimary = this.renderPrimaryEmbeddables(primaryEmbeddables, vertical, primaryFirst && !vertical, pinOffSet);
+    const renderSecondary = this.renderSecondaryEmbeddables(secondaryEmbeddables, totalPreviousQuestions, !primaryFirst && !vertical);
+    const [first, second] = primaryFirst
+                            ? [renderPrimary, renderSecondary]
+                            : [renderSecondary, renderPrimary];
+
+    return (
+      <div className={`page-content ${fullWidth ? "full" : ""}`} data-cy="page-content">
+        <div className="name">{page.name}</div>
+        <div className="introduction" dangerouslySetInnerHTML={{ __html: page.text}}></div>
```
Comment (kswenson) [inline src/components/activity-page/activity-page-content.tsx:52]: I started out using `dangerouslySetInnerHTML` in the rich text work but changed to a less dangerous approach upon code review. You should be able to crib the `renderHTML()` utility function from [this commit](https://github.com/concord-consortium/question-interactives/pull/33/commits/1b7ee9a0c101dcf67f351f7638c7342e51790915) instead. You're welcome. 😉

File: src/components/activity-page/activity-page-content.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  public componentWillUnmount() {
+    const el = document.querySelector("#app");
+    if (el) {
+      el.removeEventListener("scroll", this.handleScroll, false);
+    }
+  }
+
+  private handleScroll = (e: MouseEvent) => {
+    if (this.divRef) {
+      const scrollOffset = this.divRef.getBoundingClientRect().top < 20 ? 20 - this.divRef.getBoundingClientRect().top : 0;
```
Comment (kswenson) [inline src/components/activity-page/activity-page-content.tsx:81]: Use named constant instead of magic number.

File: src/components/app.tsx [hunk 5]
```diff
@@ -33,55 +31,63 @@ export class App extends React.PureComponent<IProps, IState> {
   }
 
   private renderActivity = () => {
+    const { currentPage } = this.state;
+    let totalPreviousQuestions = 0;
+
+    for (let page = 0; page < (currentPage - 1); page++) {
```
Comment (kswenson) [inline src/components/app.tsx:37]:
```suggestion
    for (let page = 0; page < currentPage - 1; page++) {
```

File: src/components/app.tsx [hunk 6]
```diff
@@ -33,55 +31,63 @@ export class App extends React.PureComponent<IProps, IState> {
...
+      for (let embeddable = 0; embeddable < sampleActivity.pages[page].embeddables.length; embeddable++) {
+        if (!sampleActivity.pages[page].embeddables[embeddable].section) {
+          totalPreviousQuestions++;
+        }
+      }
+    }
+
+    const fullWidth = currentPage!==0 && (sampleActivity.pages[currentPage - 1].layout === "l-responsive");
```
Comment (kswenson) [inline src/components/app.tsx:45]:
```suggestion
    const fullWidth = (currentPage !== 0) && (sampleActivity.pages[currentPage - 1].layout === "l-responsive");
```
consistency

File: src/components/activity-header/activity-nav-header.scss (hunk 1 above)
Comment (mklewandowski) [reply src/components/activity-header/activity-nav-header.scss]: Fixed.

File: src/components/activity-page/activity-page-content.tsx (hunk 2 above)
Comment (mklewandowski) [reply src/components/activity-page/activity-page-content.tsx:37]: Agreed.  The existing layout names don't, in my opinion, describe what they do (full width layout actually has a static, constrained width and would be better named "stacked").  I renamed the local variable and added a comment to help clarify.

File: src/components/activity-page/activity-page-content.tsx (hunk 4 above)
Comment (mklewandowski) [reply src/components/activity-page/activity-page-content.tsx]: Fixed.

File: src/components/app.tsx (hunk 5 above)
Comment (mklewandowski) [reply src/components/app.tsx]: Fixed.

File: src/components/app.tsx (hunk 6 above)
Comment (mklewandowski) [reply src/components/app.tsx]: Fixed.  And I removed the string constant.

File: src/components/activity-page/activity-page-content.tsx (hunk 3 above)
Comment (mklewandowski) [reply src/components/activity-page/activity-page-content.tsx]: This is sweet!  I was going to bring this topic up in a near-future stand-up.  I hate using dangerouslySetInnerHTML.  Seems so clunky.  This is a great solution! Fixed.

Comment (mklewandowski) [reply src/components/activity-page/activity-page-content.tsx]: Changed to @types/dompurify to dev dependency.

**Comment by mklewandowski** [comment]
@kswenson changed to dev dependency.  Thanks!

**Pull request merged**